clients/src/test/java/org/apache/kafka/common/requests/RequestResponseTest.java

```diff
@@ -102,7 +102,8 @@ public void testSerialization() throws Exception {
 
         checkSerialization(createUpdateMetadataRequest(0, null), 0);
         checkSerialization(createUpdateMetadataRequest(0, null).getErrorResponse(0, new UnknownServerException()), 0);
-        checkSerialization(createUpdateMetadataRequest(1, "rack1"), 1);
+        checkSerialization(createUpdateMetadataRequest(1, null), 1);
```

Comment:
But our test code does:

```String rackToSerialize = (version >= 2) ? rack : null;```

So this is misleading as it's not actually testing what we think it is. Right?